internal/benchmarks/benchmarks_test.go

```diff
@@ -0,0 +1,230 @@
+	_ "github.com/inspektor-gadget/inspektor-gadget/pkg/gadgets/trace/network/tracer"
+	_ "github.com/inspektor-gadget/inspektor-gadget/pkg/gadgets/trace/oomkill/tracer"
+	_ "github.com/inspektor-gadget/inspektor-gadget/pkg/gadgets/trace/open/tracer"
+	_ "github.com/inspektor-gadget/inspektor-gadget/pkg/gadgets/trace/signal/tracer"
+	_ "github.com/inspektor-gadget/inspektor-gadget/pkg/gadgets/trace/sni/tracer"
+	_ "github.com/inspektor-gadget/inspektor-gadget/pkg/gadgets/trace/tcp/tracer"
+	_ "github.com/inspektor-gadget/inspektor-gadget/pkg/gadgets/trace/tcpconnect/tracer"
+	_ "github.com/inspektor-gadget/inspektor-gadget/pkg/gadgets/traceloop/tracer"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/logger"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/operators"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/params"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/runtime/local"
+
+	utilstest "github.com/inspektor-gadget/inspektor-gadget/internal/test"
+	containercollection "github.com/inspektor-gadget/inspektor-gadget/pkg/container-collection"
+)
+
+type Attacher interface {
+	AttachContainer(container *containercollection.Container) error
+	DetachContainer(*containercollection.Container) error
+}
+
+type TestOperator struct{}
+
+type TestOperatorContext string
+
+func (t *TestOperator) Name() string {
+	return "TestOperator"
+}
+
+func (t *TestOperator) Description() string {
+	return "TestOperator allows to run tests with many fake containers"
+}
+
+func (t *TestOperator) GlobalParamDescs() params.ParamDescs {
+	return nil
+}
+
+func (t *TestOperator) ParamDescs() params.ParamDescs {
+	return nil
+}
+
+func (t *TestOperator) Dependencies() []string {
+	return nil
+}
+
+func (t *TestOperator) CanOperateOn(desc gadgets.GadgetDesc) bool {
+	// Create a new dummy instance
+	dummyInstance, err := desc.(gadgets.GadgetInstantiate).NewInstance()
+	if err != nil {
+		return false
+	}
+
+	// Check whether gadget supports attaching
+	_, ok := dummyInstance.(Attacher)
+	return ok
+}
+
+func (t *TestOperator) Init(params *params.Params) error {
+	return nil
+}
+
+func (t *TestOperator) Close() error {
+	return nil
+}
+
+func (t *TestOperator) Instantiate(gadgetContext operators.GadgetContext, gadgetInstance any, params *params.Params) (operators.OperatorInstance, error) {
+	return &TestOperatorInstance{
+		TestOperator: t, // for Name()
+		attacher:     gadgetInstance.(Attacher),
+		containers:   gadgetContext.Context().Value(TestOperatorContext("containers")).([]*containercollection.Container),
+		testingTB:    gadgetContext.Context().Value(TestOperatorContext("testing.TB")).(testing.TB),
+	}, nil
+}
+
+type TestOperatorInstance struct {
+	*TestOperator
+	attacher   Attacher
+	containers []*containercollection.Container
+	testingTB  testing.TB
+}
+
+func (t *TestOperatorInstance) PreGadgetRun() error {
+	for i := range t.containers {
+		t.attacher.AttachContainer(t.containers[i])
+	}
+	return nil
```

Comment:
I'm wondering if this should also create a fake mntns map and call `SetMountNsMap` on the gadgets that support it. I know the cost of updating an eBPF map is much lower than the cost of attaching a new program, however to make the comparison fair I think we should also make it.

Reply:
Done